src/main/scala/chisel3/simulator/package.scala

```diff
@@ -32,7 +32,11 @@ package object simulator {
     private[simulator] val wrapped: T,
     private[simulator] val ports:   Seq[(Data, ModuleInfo.Port)],
     private[simulator] val layers:  Seq[chisel3.layer.Layer]
-  )
+  ) {
+
+    private[chisel3] lazy val portMap = ports.toMap
```

Comment:
Why lazy? do we ever not want it?

Reply:
I had originally thought no. However, given that this will almost always execute via the macro control paths, we should just change this to be non-lazy. I noticed I can also narrow it be private to the `simulator` package, too. Both changes are applied in the rebase.